L1Trigger/TrackFindingTMTT/interface/L1fittedTrack.h

```diff
@@ -0,0 +1,416 @@
+          (this->zAtChosenR() > secTmp_->zAtChosenR_Min() && this->zAtChosenR() < secTmp_->zAtChosenR_Max());
+      return (insidePhi && insideEta);
+    }
+
+    // Digitize track and degrade helix parameter resolution according to effect of digitisation.
+    void digitizeTrack(const std::string& fitterName);
+
+    // Access to detailed info about digitized track. (Gets nullptr if trk not digitized)
+    const DigitalTrack* digitaltrack() const { return digitalTrack_.get(); }
+
+  private:
+    //--- Configuration parameters
+    const Settings* settings_;
+
+    //--- The 3D hough-transform track candidate which was fitted.
+    const L1track3D* l1track3D_;
+
+    //--- The stubs on the fitted track (can differ from those on HT track if fit kicked off stubs with bad residuals)
+    std::vector<Stub*> stubs_;
+    std::vector<const Stub*> stubsConst_;
+    unsigned int nLayers_;
+
+    //--- Bit-encoded hit pattern (where layer number assigned by increasing distance from origin, according to layers track expected to cross).
+    unsigned int hitPattern_;
+
+    //--- The fitted helix parameters and fit chi-squared.
+    float qOverPt_;
+    float d0_;
+    float phi0_;
+    float z0_;
+    float tanLambda_;
+    float chi2rphi_;
+    float chi2rz_;
+
+    //--- Ditto with beam-spot constraint applied in case of 5-parameter fit, plus boolean to indicate
+    bool done_bcon_;
+    float qOverPt_bcon_;
+    float d0_bcon_;
+    float phi0_bcon_;
+    float chi2rphi_bcon_;
+
+    //--- The number of helix parameters being fitted (=5 if d0 is fitted or =4 if d0 is not fitted).
+    unsigned int nHelixParam_;
+
+    //--- Phi sector and eta region used track finding code that this track was in.
+    unsigned int iPhiSec_;
+    unsigned int iEtaReg_;
+    //--- Opto-link ID from HT to Track Fitter.
+    unsigned int optoLinkID_;
+
+    //--- Information about its association (if any) to a truth Tracking Particle.
+    const TP* matchedTP_;
+    std::vector<const Stub*> matchedStubs_;
+    unsigned int nMatchedLayers_;
+
+    //--- Has the track fit declared this to be a valid track?
+    bool accepted_;
+
+    //--- Sector class used to check if fitted track trajectory is in same sector as HT used to find it.
+    std::shared_ptr<Sector> secTmp_;  // shared so as to allow copy of L1fittedTrack.
+    //--- r-phi HT class used to determine HT cell location that corresponds to fitted track helix parameters.
+    std::shared_ptr<HTrphi> htRphiTmp_;
+
+    //--- Info specific to KF fitter.
+    unsigned int nSkippedLayers_;
+    unsigned int numUpdateCalls_;
+    //--- Info specific to LR fitter.
+    int numIterations_;
+    std::string lostMatchingState_;
+    std::unordered_map<std::string, int> stateCalls_;
+
+    std::shared_ptr<DigitalTrack> digitalTrack_;  // Class used to digitize track if required.
+
+    bool consistentCell_;
+
+    static const std::vector<Stub*> noStubs_;  // Empty vector used to initialize rejected tracks.
```

Comment:
remaining static variable

Reply:
Since "static const" data is thread safe, and following the discussion leading to https://github.com/cms-sw/cmssw/pull/29381#issuecomment-628889782 , we propose to leave this code unchanged.